libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -43,6 +46,7 @@ fun NavigationRoute.refreshRoute(
     legAnnotations: List<LegAnnotation?>?,
     incidents: List<List<Incident>?>?,
     closures: List<List<Closure>?>?,
+    unrecognizedProperties: Map<String, JsonElement>?,
```

Comment:
```suggestion
    waypoints: List<DirectionsWaypoint>?,
```
This makes more sense from the perspective that anything could be refreshed via unregonized properties in the future and passing just the map gives no indication to this function what in the response should that map actually impact.